scripts/process_uploads.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python3
+"""
+Processes files uploaded to gcp
+"""
+
+import os
+import csv
+from os.path import join, basename
+import subprocess
+import tempfile
+import shutil
+from typing import List
+import hail as hl
+import hailtop.batch as hb
+from joint_calling.upload_processor import batch_move_files
+
+
+def samples_from_csv(local_csv_path):
+    """ Determines list of samples from a given csv file. """
+
+    samples: List[str] = []
+    with open(local_csv_path) as csv_file:
+        csv_reader = csv.DictReader(csv_file)
+        for row in csv_reader:
+            samples.append(row['sample.sample_name'])
+
+    return samples
+
+
+def determine_samples(upload_bucket: str, previous_batch_path: str):
+    """Determine files that should be moved to main vs archive bucket.
+    Determines the difference between the latest CSV file within upload
+    and the CSV file from the most recent batch as the list of samples to
+    be moved.
+
+    Assumptions
+    ==========
+    Only one .csv file exists in the upload bucket and previous batch directory.
+    """
+    local_tmp_dir = tempfile.mkdtemp()
+    all_samples: List[str] = []
+    previous_samples: List[str] = []
+
+    # Pull the samples listed in the most recent CSV file
+    cmd = f'gsutil ls \'gs://{upload_bucket}/*.csv\''
+    curr_csv_file_path = subprocess.check_output(cmd, shell=True).decode().strip()
```

Comment:
It's easier and faster to do this using the https://pypi.org/project/google-cloud-storage/ client libraries.

Reply:
I've updated the function now. It still uses subprocess to find a .csv file. I couldn't find an option to specify a particular file extension or wildcard within the find_blob function. Is there a better way to go about this?

Reply:
Oh, just saw your comment now. Unfortunately you can only filter by prefix, but you can then just filter the results using `endswith`.

Reply:
Thanks! Have fixed :)